File: DSA/Ayush_Mishra_590014844/Day_54_Question_3.c
#include <stdio.h>

int main() {
    int n, k;
    scanf("%d %d", &n, &k);
    int nums[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &nums[i]);
    }

    int count = 0, left = 0;
    long long prod = 1;
    for (int right = 0; right < n; right++) {
        prod *= nums[right];
        while (left <= right && prod >= k) {
            prod /= nums[left];
            left++;
        }
        count += right - left + 1;
    }

    printf("%d\n", count);
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_54_Question_3.py
def num_subarray_product_less_than_k(nums, k):
    if k <= 1:
        return 0
    prod = 1
    left = 0
    count = 0
    for right in range(len(nums)):
        prod *= nums[right]
        while prod >= k:
            prod /= nums[left]
            left += 1
        count += right - left + 1
    return count

n, k = map(int, input().split())
nums = list(map(int, input().split()))
print(num_subarray_product_less_than_k(nums, k))
